### ISSUE TITLE
404 / Timeout Error

### ISSUE THREAD

**Issue by SebastianGiro** [GatoLoko/Wattpad2Epub#10]
After the script run for about 5 minutes, at some point, I had a 404 timeout:

```python
Traceback (most recent call last):
  File "wattpad2epub.py", line 271, in <module>
    get_book(args.initial_url[0])
  File "wattpad2epub.py", line 228, in get_book
    chapter = get_chapter("{}{}".format(base_url, item['href']))
  File "wattpad2epub.py", line 135, in get_chapter
    for j in get_page(page_url):
  File "wattpad2epub.py", line 118, in get_page
    text = get_html(text_url).select('pre')
  File "wattpad2epub.py", line 63, in get_html
    request = urllib.request.urlopen(req)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 564, in error
    result = self._call_chain(*args)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 756, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
```

What should happen?
Just keep trying until I cancel it, or at least don't lose all the things the script did in the process, so we can resume it. (This error happened like 4 times in 10 tries)

**Comment by SebastianGiro** [comment]
I created this pull request, this should help to clarify a little how to run the script and what know where the output will be at least https://github.com/GatoLoko/Wattpad2Epub/pull/11

**Comment by GatoLoko** [comment]
A 404 error isn't a timeout, but a "page not found" or "this page doesn't exist in the server" error.

If it's the initial page, it means the URL provided by the user is wrong. This could be verified before starting to do anything.

If it's any other URL, it may be either a change in the server (temporal or not) or a bug in the script, so this is harder to handle.

I'll add code to handle the 404 error, "hidding" all this messages by default, unless we are using the -d parameter for debugging, and replacing them with:

```
HTTP Error 404: Not Found    
Aborting...
```
About retrying... It already retries up to 10 times before giving up.

Resuming isn't supported right now, at all. The output file only gets written to disk IF everything goes well. This implies a lot more code than just handling an error, and should be discussed on its own issue. #12

**Comment by sebghtray** [comment]
I will try to check it this weekend.

The 404 was really weird, because it happened like in the chapter 40 of 80.
And after some tries it finished properly.

**Comment by GatoLoko** [comment]
> The 404 was really weird, because it happened like in the chapter 40 of 80.
> And after some tries it finished properly.

If once the server stopped causing problems, the program finished properly, it means the bug  was NOT in Wattpad2Epub, but on  the server side, and that's not something I can fix.
